src/v/rpc/transport.cc

```diff
@@ -80,7 +80,11 @@ ss::future<> transport::connect(clock_type::duration connection_timeout) {
 }
 
 void transport::reset_state() {
-    _correlation_idx = 0;
+    /*
+     * the _correlation_idx is explicitly not reset as a pragmatic approach to
+     * dealing with an apparent race condition in which two requests with the
+     * same correlation id are in flight shortly after a reset.
+     */
```

Comment:
Nice! This fix seems fine for this issue with reused correlation IDs, though I wonder what surprises could be lurking from this race in the rest of the reset fields.

Reply:
yeh it's a concern for sure and i think we should continue to look for the problem.